app/components/message_groups/message_groups.js

```diff
@@ -0,0 +1,20 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+  connect() {
+    document.addEventListener('turbo:frame-render', event => {
+      const anchor = new URL(document.location).hash;
+      if (!anchor) return;
+
+      const turboFrame = event.target;
+      const elementToScrollTo = turboFrame.querySelector(anchor);
+      if (!elementToScrollTo) return;
+
+      elementToScrollTo.scrollIntoView({ behavior: 'smooth', block: 'center' });
+      Object.assign(elementToScrollTo.style, {
+        borderColor: '#007eff',
+        boxShadow: '0 0 0 3px #7ac8ff',
+      });
```

Comment:
This was added to mimic the styles in `ChatMessage:target`. When relying the native browser functionality, the `:target` pseudo class applies the styles to the element being targeting in the url's anchor. When using js, this does not work, so we need to apply the styles with js.

See https://developer.mozilla.org/en-US/docs/Web/CSS/:target

Maybe we should remove the CSS since it would never used(?) Maybe, for development, by toggling the `:target` pseudo class in the browser tools.

Reply:
In this case we really should add an appropriate comment into the code to explain exactly that. 

Side note: Maybe it's good practice to put inline code whenever an explanatory comments seems needed in the PR. Same as if one feels the need to put an inline comment it's better to refactor the code ..